openmdao/core/problem.py

```diff
@@ -103,6 +103,8 @@ def __init__(self, model=None, comm=None, use_ref_vector=True, root=None):
 
         self._mode = None  # mode is assigned in setup()
 
+        recording_iteration_stack = []
+
```

Comment:
So, I did try this when I found the test failure problem, and it didn't fix it for me. Do you maybe have access to a linux system that you can test it on?